src/full/Agda/TypeChecking/Serialise/Instances.hs

```diff
@@ -28,8 +28,5 @@ instance EmbPrj Interface where
       icodeN' interface a b c (fromImportedModules d) e f g h i j k l m n o p q r s t u v
     where interface a b c = Interface a b c . toImportedModules
 
-  value = vcase valu where
-    valu [a, b, c, d, e, f, g, h, i, j, k, l, m, n, o, p, q, r, s, t, u, v] =
-        valuN interface a b c d e f g h i j k l m n o p q r s t u v
-      where interface a b c = Interface a b c . toImportedModules
-    valu _ = malformed
+  value = valueN interface
+    where interface a b c = Interface a b c . toImportedModules
```

Comment:
Is this change related to the topic of the PR or is it just some simplification you noticed while working on the code?  I fail to see the connection...

Reply:
There are more occurrences of such refactorings.
You seem to have simplified the implementation of `value` for single-constructor data types by skipping `vcase`.

Reply:
`valueN` still uses `vcase` under the hood (so should still have the same memoization/sharing behaviour), I just find it forms a more natural dual to `icodeN'`. Do appreciate it is a separate change to the enum one though, so happy to revert!

Reply:
I welcome this change.  
Could you put it in a separate commit with a descriptive commit message?  Then we would merge this with two separate commits.  This is my suggestion.